pkgs/tools/misc/topgrade/default.nix

```diff
@@ -18,7 +18,10 @@ rustPlatform.buildRustPackage rec {
     sha256 = "sha256-9zP+rWhaK4fC2Qhd0oq9WVvCkvryooYo09k7016Rbxw=";
   };
 
-  cargoSha256 = "sha256-otn0XvZ0wufD+4mCGSM0hevKM+wWSvFVCKtTu/5m1uA=";
+  cargoPatches = lib.optionals stdenv.isDarwin [ ./darwin-cargo-lock.patch ];
+  cargoSha256 =
+    if stdenv.isDarwin then "sha256-rkcEF/INNVn9K4p0/1M++l6lnjtZp1Srx57gkaqcKek="
+    else "sha256-otn0XvZ0wufD+4mCGSM0hevKM+wWSvFVCKtTu/5m1uA=";
```

Comment:
When we make the patch not conditional we don't need this. This is also ugly to update.